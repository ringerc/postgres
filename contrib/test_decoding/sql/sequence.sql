SET synchronous_commit = on;

SELECT 'init' FROM pg_create_logical_replication_slot('regression_slot', 'test_decoding');

CREATE SEQUENCE test_seq;

SELECT sum(nextval('test_seq')) FROM generate_series(1, 100);

ALTER SEQUENCE test_seq RESTART WITH 2;

SELECT sum(nextval('test_seq')) FROM generate_series(1, 100);

ALTER SEQUENCE test_seq INCREMENT BY 5;

SELECT sum(nextval('test_seq')) FROM generate_series(1, 100);

ALTER SEQUENCE test_seq CACHE 10000;

SELECT sum(nextval('test_seq')) FROM generate_series(1, 15000);

DROP SEQUENCE test_seq;

SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-xids', '0', 'skip-empty-xacts', '1', 'include-sequences', '1');


CREATE SEQUENCE test2_seq MINVALUE 100 MAXVALUE 200 START 200 INCREMENT BY -1 CYCLE;

SELECT sum(nextval('test2_seq')) FROM generate_series(1, 300);

SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-xids', '0', 'skip-empty-xacts', '1', 'include-sequences', '1');

DROP SEQUENCE test2_seq;


SELECT pg_drop_replication_slot('regression_slot');
