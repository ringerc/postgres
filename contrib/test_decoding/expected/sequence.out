SET synchronous_commit = on;
SELECT 'init' FROM pg_create_logical_replication_slot('regression_slot', 'test_decoding');
 ?column? 
----------
 init
(1 row)

CREATE SEQUENCE test_seq;
SELECT sum(nextval('test_seq')) FROM generate_series(1, 100);
 sum  
------
 5050
(1 row)

ALTER SEQUENCE test_seq RESTART WITH 2;
SELECT sum(nextval('test_seq')) FROM generate_series(1, 100);
 sum  
------
 5150
(1 row)

ALTER SEQUENCE test_seq INCREMENT BY 5;
SELECT sum(nextval('test_seq')) FROM generate_series(1, 100);
  sum  
-------
 35350
(1 row)

ALTER SEQUENCE test_seq CACHE 10000;
SELECT sum(nextval('test_seq')) FROM generate_series(1, 15000);
    sum    
-----------
 571552500
(1 row)

DROP SEQUENCE test_seq;
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-xids', '0', 'skip-empty-xacts', '1', 'include-sequences', '1');
                 data                 
--------------------------------------
 sequence test_seq advanced to 1
 sequence test_seq advanced to 33
 sequence test_seq advanced to 66
 sequence test_seq advanced to 99
 sequence test_seq advanced to 132
 sequence test_seq advanced to 2
 sequence test_seq advanced to 34
 sequence test_seq advanced to 67
 sequence test_seq advanced to 100
 sequence test_seq advanced to 133
 sequence test_seq advanced to 101
 sequence test_seq advanced to 266
 sequence test_seq advanced to 431
 sequence test_seq advanced to 596
 sequence test_seq advanced to 761
 sequence test_seq advanced to 601
 sequence test_seq advanced to 50761
 sequence test_seq advanced to 100761
(18 rows)

CREATE SEQUENCE test2_seq MINVALUE 100 MAXVALUE 200 START 200 INCREMENT BY -1 CYCLE;
SELECT sum(nextval('test2_seq')) FROM generate_series(1, 300);
  sum  
-------
 45147
(1 row)

SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-xids', '0', 'skip-empty-xacts', '1', 'include-sequences', '1');
                 data                 
--------------------------------------
 sequence test_seq advanced to 1
 sequence test_seq advanced to 33
 sequence test_seq advanced to 66
 sequence test_seq advanced to 99
 sequence test_seq advanced to 132
 sequence test_seq advanced to 2
 sequence test_seq advanced to 34
 sequence test_seq advanced to 67
 sequence test_seq advanced to 100
 sequence test_seq advanced to 133
 sequence test_seq advanced to 101
 sequence test_seq advanced to 266
 sequence test_seq advanced to 431
 sequence test_seq advanced to 596
 sequence test_seq advanced to 761
 sequence test_seq advanced to 601
 sequence test_seq advanced to 50761
 sequence test_seq advanced to 100761
 sequence test2_seq advanced to 200
 sequence test2_seq advanced to 168
 sequence test2_seq advanced to 135
 sequence test2_seq advanced to 102
 sequence test2_seq advanced to 100
 sequence test2_seq advanced to 168
 sequence test2_seq advanced to 135
 sequence test2_seq advanced to 102
 sequence test2_seq advanced to 100
 sequence test2_seq advanced to 168
 sequence test2_seq advanced to 135
 sequence test2_seq advanced to 102
(30 rows)

DROP SEQUENCE test2_seq;
SELECT pg_drop_replication_slot('regression_slot');
 pg_drop_replication_slot 
--------------------------
 
(1 row)

